File: Структуры данных/set.py
def get_negative_elements_only_from_b(set_a, set_b):
    result = set()
    for element in set_b:
        if element < 0 and element not in set_a:
            result.add(element)
    return result



set_a = set(map(int, input("Введите множество A: ").split()))
set_b = set(map(int, input("Введите множество B: ").split()))

result = get_negative_elements_only_from_b(set_a, set_b)
print("Результат:", result)

File: Структуры данных/set.cpp
#include <iostream>
#include <set>

using namespace std;

set<int> get_negative_elements_only_from_b(set<int>* set_a, set<int>* set_b) {
    set<int> result;
    for (int element : *set_b) {
        if (element < 0 && set_a->find(element) == set_a->end()) {
            result.insert(element);
        }
    }
    return result;
}

int main() {
    set<int> set_a, set_b;
    int x;

    cout << "Введите множество A: ";
    while (cin >> x) {
        set_a.insert(x);
        if (cin.peek() == '\n') break;
    }

    cout << "Введите множество B: ";
    while (cin >> x) {
        set_b.insert(x);
        if (cin.peek() == '\n') break;
    }

    auto result = get_negative_elements_only_from_b(&set_a, &set_b);

    cout << "Результат: ";
    for (int elem : result) {
        cout << elem << " ";
    }
    cout << endl;

    return 0;
}
